contracts/01-tamagotchi/tests/smoke.rs

```diff
@@ -7,4 +8,15 @@ fn smoke_test() {
     let _program = Program::current(&sys);
 
     // TODO: 8️⃣ Test the program initialization and message handling
+    let res = _program.send(2, String::from("Tamagotchi Name"));
+    assert!(!res.main_failed());
+
+    let res = _program.send(2, TmgAction::Name);
+    let expected_log = Log::builder()
+        .dest(2)
+        .payload(TmgEvent::Name(String::from("Tamagotchi Name")));
+    assert!(res.contains(&expected_log));
+
+    let res = _program.send(2, TmgAction::Age);
+    assert!(!res.log().is_empty());
```

Comment:
You could write a more expressive test by using the [`System::spend_blocks`](https://docs.gear.rs/gtest/struct.System.html#method.spend_blocks) function.